### PR TITLE
[feature](window-func) support percent_rank window function

### LINKED ISSUE #30636: [Feature] support pecent_rank window function
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

mysql和postgresql提供该函数，常用于按百分比过滤数据，例如取销售额前20%的商品

https://www.mysqltutorial.org/mysql-window-functions/mysql-percent_rank-function/
https://www.postgresqltutorial.com/postgresql-window-function/postgresql-percent_rank-function/

### Use case

PERCENT_RANK() OVER (
    [PARTITION BY partition_expression, ... ]
    ORDER BY sort_expression [ASC | DESC], ...
)

### Related issues

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by sjyango**
## Proposed changes

Issue Number: close #30636

support percent_rank window function for doris.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by sjyango** [comment]
run pipelinex_p0

**Comment by sjyango** [comment]
run p0

**Comment by sjyango** [comment]
run p0

**Comment by sjyango** [comment]
run pipelinex_p0

**Comment by sjyango** [comment]
run p0

**Comment by sjyango** [comment]
run pipelinex_p0

**Comment by sjyango** [comment]
run p0

**Comment by sjyango** [comment]
run pipelinex_p0

**Review by morrySnow — DISMISSED** [review]

**Comment by morrySnow** [comment]
run buildall

**Comment by sjyango** [comment]
run pipelinex_p0

**Review by zclllyybb — APPROVED** [review]
LGTM

File: be/src/vec/aggregate_functions/aggregate_function_window.h [hunk 1]
```diff
@@ -174,6 +175,62 @@ class WindowFunctionDenseRank final
...
+                                Arena* arena) const override {
+        int64_t peer_group_count = frame_end - frame_start;
+        if (WindowFunctionPercentRank::data(place).peer_group_start != frame_start) {
+            WindowFunctionPercentRank::data(place).peer_group_start = frame_start;
+            WindowFunctionPercentRank::data(place).rank +=
+                    WindowFunctionPercentRank::data(place).count;
+        }
+        WindowFunctionPercentRank::data(place).count = peer_group_count;
+        // some variables are partition related, but there is no chance to init them
+        // when the new partition arrives, so we calculate them evey time now.
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_window.h:213]: `every time`

File: be/src/vec/aggregate_functions/aggregate_function_window.h [hunk 2]
```diff
@@ -174,6 +175,62 @@ class WindowFunctionDenseRank final
...
+    int64_t rank = 0;
+    int64_t count = 0;
+    int64_t peer_group_start = 0;
+    int64_t partition_size = 0;
+};
+
+class WindowFunctionPercentRank final
+        : public IAggregateFunctionDataHelper<PercentRankData, WindowFunctionPercentRank> {
+private:
+    inline static double _cal_percent(int64 rank, int64 total_rows) {
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_window.h:188]: remove the `inline`

File: be/src/vec/aggregate_functions/aggregate_function_window.h [hunk 3]
```diff
@@ -174,6 +175,62 @@ class WindowFunctionDenseRank final
...
+        int64_t peer_group_count = frame_end - frame_start;
+        if (WindowFunctionPercentRank::data(place).peer_group_start != frame_start) {
+            WindowFunctionPercentRank::data(place).peer_group_start = frame_start;
+            WindowFunctionPercentRank::data(place).rank +=
+                    WindowFunctionPercentRank::data(place).count;
+        }
+        WindowFunctionPercentRank::data(place).count = peer_group_count;
+        // some variables are partition related, but there is no chance to init them
+        // when the new partition arrives, so we calculate them evey time now.
+        WindowFunctionPercentRank::data(place).partition_size = partition_end - partition_start;
```
Comment (HappenLee) [inline be/src/vec/aggregate_functions/aggregate_function_window.h:214]: `if (WindowFunctionPercentRank::data(place).peer_group_start != frame_start)` maybe means new partition arrives？

**Comment by sjyango** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Pull request merged**